front-end/src/screens/AddressLoginForm/index.tsx

```diff
@@ -0,0 +1,224 @@
+			setAddress(accounts[0].address);
+		}
+
+		return;
+	};
+
+	const onAccountChange = (event: React.SyntheticEvent<HTMLElement, Event>, data: DropdownProps) => {
+		const addressValue = data.value as string;
+		setAddress(addressValue);
+	};
+
+	const navigateLogin = () => {
+		history.push('/login');
+	};
+
+	const handleLogin = async () => {
+		if (!accounts.length) {
+			return getAccounts();
+		}
+
+		try {
+			const injected = await web3FromSource(accounts[0].meta.source);
+			const signRaw = injected && injected.signer && injected.signer.signRaw;
+
+			if (!signRaw) {
+				return console.error('Signer not available');
+			}
+
+			const { data: startResult } = await addressLoginStartMutation({
+				variables: {
+					address
+				}
+			});
+
+			const signMessage = startResult?.addressLoginStart?.signMessage;
+
+			if (!signMessage) {
+				throw new Error('Challenge message not found');
+			}
+
+			const { signature } = await signRaw({
+				address,
+				data: stringToHex(signMessage),
+				type: 'bytes'
+			});
+
+			const { data: loginResult } = await addressLoginMutation({
+				variables: {
+					address,
+					signature
+				}
+			});
+
+			if (loginResult?.addressLogin?.token && loginResult?.addressLogin?.user) {
+				handleLoginUser({ token: loginResult.addressLogin.token, user: loginResult.addressLogin.user }, currentUser);
+				history.push('/');
+			} else {
+				throw new Error('Web3 Login failed');
+			}
+		} catch (error) {
+			console.error(error);
+			queueNotification({
+				header: 'Failed!',
+				message: cleanError(error.message),
+				status: NotificationStatus.ERROR
+			});
+		}
+	};
+
+	return (
+		<Grid className={className}>
+			<Grid.Column only='tablet computer' tablet={2} computer={4} largeScreen={5} widescreen={5}/>
+			<Grid.Column mobile={16} tablet={12} computer={8} largeScreen={6} widescreen={6}>
+				<Form onSubmit={handleLogin}>
+					<h3>Login With Web3 Account</h3>
+					{extensionNotFound?
+						<div className='card'>
+							<ExtensionNotDetected />
+						</div>
+						: null
+					}
+					{accountsNotFound?
+						<div className='card'>
+							<div className='text-muted'>You need at least one account in Polkadot-js extenstion to login.</div>
+							<div className='text-muted'>Please reload this page after adding accounts.</div>
+						</div>
+						: null
+					}
+					{accounts.length > 0 ?
+						<Form.Group>
+							<Form.Field width={16}>
+								<AccountSelectionForm
+									title='Choose linked account'
+									accounts={accounts}
+									address={address}
+									onAccountChange={onAccountChange}
+								/>
+							</Form.Field>
+						</Form.Group>
+						: <div>Loading Accounts ...</div>
+					}
+					<div className='info'>
+						Alternatively, you can login with your username
+					</div>
+					<div className={'mainButtonContainer'}>
+						<Button
+							secondary
+							onClick={navigateLogin}
+						>
+							Login with username
+						</Button>
+						<Button
```

Comment:
if `getAccounts` is done in the useEffect, this button has no more reason to be I guess.

Reply:
removed